nucypher/config/storages.py

```diff
@@ -57,17 +58,17 @@ class InvalidNodeCertificate(RuntimeError):
     def __init__(self,
                  federated_only: bool,  # TODO# 466
                  character_class=None,
-                 serializer: Callable = NODE_SERIALIZER,
-                 deserializer: Callable = NODE_DESERIALIZER,
+                 encoder: Callable = None,
```

Comment:
`Optional[Callable]` maybe? Or use class vars as defaults like it was in the original?

Reply:
Actually, are these parameters ever used at all?

Reply:
Class var as defaults actually don't work very well with overriding. I was puzzled initially because I first did the change to overriding the encoders to the identity, but the serialized version was still hex strings; it was like it wasn't using the overridden version at all. I'm not sure what's the explanation to be honest, but referring to `self.NODE_SERIALIZED`  inside the `__init__` did the trick.

I'm fine with `Optional[Callable]`. We definitely need more type hinting. 

By the way, if you want to iterate over how encoding works, feel free to directly push to this branch!

Reply:
Perhaps we don't even need these parameters at all. I couldn't find a place in the codebase where they would be used.

Reply:
Yeah, you're right. This is an older part of the codebase, and probably the idea back in the time was to make it more generic, but if after 2 years we are not using it, I think it's fine to change it. I don't think I can work on this until tomorrow, so if you have the itch, be my guest :). If not, I'll continue from here tomorrow

Reply:
@cygnusv - Summed it up well - Only built in mind with extensibility - It was ( is? ) unclear how dapp builders may want to store Ursulas.

Inside nucypher itself - i'ts nearly written in stone.

Reply:
Do you think it is worth preserving as a part of the public API? In that case it needs to be documented and tested (not necessarily in this PR)

Reply:
I'm leaning towards No, but my mind can easily be changed.

Reply:
✅
I just got rid of it